projects/plugins/jetpack/modules/subscriptions.php

```diff
@@ -989,6 +992,189 @@ public function register_post_meta() {
 		register_meta( 'post', '_jetpack_post_was_ever_published', $jetpack_post_was_ever_published );
 	}
 
+	/**
+	 * Store the list of subscribers when a post is first emailed.
+	 *
+	 * This method is called when a post is published and emails are sent to subscribers.
+	 * It stores the subscriber count and metadata in post meta for debugging purposes.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param int     $post_ID Post ID.
+	 * @param array   $flags   Post flags including send_subscription.
+	 * @param WP_Post $post    Post object.
+	 *
+	 * @return void
+	 */
+	public function store_subscribers_when_sent( $post_ID, $flags, $post ) {
+		// Only store if emails are being sent.
+		if ( ! isset( $flags['send_subscription'] ) || ! $flags['send_subscription'] ) {
+			return;
+		}
+
+		// Only store once - check if we've already stored subscribers for this post.
+		$existing_subscribers = get_post_meta( $post_ID, '_jetpack_newsletter_subscribers_when_sent', true );
+		if ( ! empty( $existing_subscribers ) ) {
+			return;
+		}
+
+		// Only store for posts.
+		if ( 'post' !== $post->post_type ) {
+			return;
+		}
+
+		// Fetch subscriber data from WordPress.com API.
+		$subscriber_data = $this->get_subscriber_data();
+
+		// Store subscriber data with timestamp.
+		$data_to_store = array(
+			'timestamp'         => current_time( 'mysql' ),
+			'email_subscribers' => isset( $subscriber_data['email_subscribers'] ) ? (int) $subscriber_data['email_subscribers'] : 0,
+			'paid_subscribers'  => isset( $subscriber_data['paid_subscribers'] ) ? (int) $subscriber_data['paid_subscribers'] : 0,
+			'all_subscribers'   => isset( $subscriber_data['all_subscribers'] ) ? (int) $subscriber_data['all_subscribers'] : 0,
+			'subscriber_list'   => isset( $subscriber_data['subscriber_list'] ) && is_array( $subscriber_data['subscriber_list'] ) ? $subscriber_data['subscriber_list'] : array(),
+		);
+
+		update_post_meta( $post_ID, '_jetpack_newsletter_subscribers_when_sent', $data_to_store );
+	}
+
+	/**
+	 * Get subscriber data from WordPress.com API.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @return array Subscriber data including counts and email list.
+	 */
+	private function get_subscriber_data() {
+		$subscriber_data = array(
+			'email_subscribers' => 0,
+			'paid_subscribers'  => 0,
+			'all_subscribers'   => 0,
+			'subscriber_list'   => array(),
+		);
+
+		// Only fetch if Jetpack is connected.
+		if ( ! Jetpack::is_connection_ready() ) {
+			return $subscriber_data;
+		}
+
+		$site_id = Jetpack_Options::get_option( 'id' );
+
+		// First, get subscriber counts from stats endpoint.
+		$stats_path     = sprintf( '/sites/%d/subscribers/stats', $site_id );
+		$stats_response = Client::wpcom_json_api_request_as_blog(
```

Comment:
Same like for the other API call, I think we should save that data in a transient to save outgoing calls when possible.